rubyforgood/casa
# [#1741] Generate past court reports from past court date

### What github issue is this PR for, if any?
Resolves #1741

### What changed, and why?
Admins, supervisors and case volunteers are now able to download a .docx report of a given past court date.

### How will this affect user permissions?
The permissions to download the .docx reports are the same as the ones to see the past court date itself (since it is the same action in the controller).

- Volunteer permissions: Can download the report if they are associated with the case
- Supervisor permissions: Can only download the report if they are in the same organization as the case
- Admin permissions: Can always download the report

### How is this tested? (please write tests!) 💖💪
Automated specs:
- Updated past court date request specs to see if it downloads the .docx
- Updated past court date view specs to check if the Download button is visible
- Created past court date model specs

Manual testing:
- Create some past court dates, with (and without) court details
- Go to the page, download the report and check if everything is displayed properly

### Screenshots please :)
#### A new button is now present at the past court date page
![1](https://user-images.githubusercontent.com/72531802/120543193-6e81bc80-c3c2-11eb-8a35-461514d42fd9.png)

#### Desktop view:
![2](https://user-images.githubusercontent.com/72531802/120543199-6fb2e980-c3c2-11eb-8688-0a78e237a0e9.png)

#### Mobile view:
![3](https://user-images.githubusercontent.com/72531802/120543206-70e41680-c3c2-11eb-8a56-9e15f9b66a2d.png)

#### Generated document:
![4](https://user-images.githubusercontent.com/72531802/120543209-72adda00-c3c2-11eb-9c36-5ef54f03e96e.png)




### Feelings gif (optional)
What it felt like to be forced to use Microsoft Word to generate the template:

![errors](https://media.giphy.com/media/Vu7FU5T4RJPo1esgna/giphy.gif)

Just kidding ;)
No errors occurred but I had to use someone else's computer to use MS Word
